Profiler/utils/WindowsUtils.cpp

```diff
@@ -52,6 +52,25 @@ std::vector<std::string> WindowsUtils::listEnvironmentVariables() {
 	return variables;
 }
 
+std::string WindowsUtils::getPathOfProfiler() {
+	char profilerPath[_MAX_PATH];
+
+	HMODULE hm = NULL;
+	if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
+		GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+		(LPCSTR)&getPathOfProfiler, &hm) != 0)
+	{
+		size_t length = GetModuleFileName(hm, profilerPath, _MAX_PATH);
+		if (length != 0)
+		{
+			return std::string(profilerPath, length);
+		}
+	}
+
+	// Failed to retrieve module path, try to retrieve it from the environment variable instead
+	return getConfigValueFromEnvironment("PATH");
```

Comment:
It's a bit tricky, because for the `CORECLR_` variables we recommend using `PATH_32` and `PATH_64` instead of `PATH`. 🤔 I will try to add a not-too-ugly fallback handling. :grin: